apps/admin_api/lib/admin_api/application.ex

```diff
@@ -9,6 +9,10 @@ defmodule AdminAPI.Application do
   def start(_type, _args) do
     import Supervisor.Spec
     DeferredConfig.populate(:admin_api)
+
+    settings = Application.get_env(:admin_api, :settings)
```

Comment:
Hmm, I have a feeling that `:settings` shouldn't be in config (since it's not exactly configurable). Maybe as a `@settings [...]` in here (i.e. `application.ex`) instead?

Reply:
Or if we do it might make sense to use something among the line of:

```
config :admin_url,
  sender_email: {:ewallet_config, "default"},
  something_else: {:ewallet_config, nil}
```

…and iterate over the keys. In this case we can simplify `EWalletConfig.Config.register_and_load` to just:

```
EWalletConfig.Config.register_and_load(:admin_api)
```

Reply:
Hmm, but from my point of view, it is configurable - you can set which settings you need in the app for it to work, we can add more in the future or remove them later. I can't really see how it's different from other config like `generators` or the `mime_types`. 

About listing the settings, we could do that, but I don't really like the idea of having default in there. I see the settings as a global system, with one default value for all the apps, don't really want to let each app set their own default. At least, that's how I see it with the current implementation, it might evolve and requires apps to each have different default values 🤷‍♂️

Reply:
Maybe we can use `{:ewallet_config, "key_to_retrieve"}` or something (where `nil` defaults to the key)?

My issue is mainly with having `:settings` key in config, which doesn't feel exactly right to me. As in, we're defining the values to be populated into `app_env` in a different ways than how these settings are normally populated (as in DeferredConfig, or other MFA-style configurations).

Also, if we use `{:ewallet_config, ...}` we could stub it in tests without involving `ewallet_config` at all.

Reply:
@sirn I'm not sure why we need that `{:ewallet_config, ...` part for? Also, one of the reason it's in the settings in that format is that I'm actually using the whole list of settings in the shared test helper function  there https://github.com/omisego/ewallet/blob/436-settings-system/apps/ewallet_config/test/support/config_test_helper.ex#L15.